### ISSUE TITLE
alternative to plugin

### ISSUE THREAD

**Issue by chenbekor** [ReactiveCouchbase/ReactiveCouchbase-play#48]
@mathieuancelin we would like to be able to use reactivecouchbase without a plugin.

why we need that - 

we are trying to integrate kamon into our play apps. kamon is a akka based monitoring / instrumentation tool that uses run time weaving in order to instrument akka.

the issue we are facing is that kamon will not act nicely if any actor system is created before calling kamon.start()  which happens in Global scala onStart method.

we think that reactivecouchbase plugin pre-initiate the actor system which collides with kamon.

can you please verify that and maybe provide a driver.start() method and driver.stop() instead of the plugin system. this way we can synchronize the calls => kamon.start() and then cb_driver.start()

this is urgent as it is blocking our progress with kamon. thanks!

**Comment by mathieuancelin** [comment]
Hi Chen,

one part of the issue is that the ReactiveCouchbase plugin, like any other plugin, is initialized before the `Global.onStart` function. 

You should start your Kamon instrumentation from a plugin and enable it before ReactiveCouchbase. 

Some pseudo code to do that : 

```scala
class KamonStarterPlugin(app: Application) extends Plugin {
  override lazy val enabled = true
  override def onStart() {
    // here whatever you need to do to start Kamon 
    MyKamonAPI.start()
  }
  
  override def onStop() {
    // here whatever you need to do to stop Kamon 
    MyKamonAPI.stop()
  }
}
```

then in your `conf/play.plugins` file add the following line. The number at the beginning should be smaller than the one on the ReactiveCouchbase line (to start before). You can play with that number to match your needs.

```
100:path.to.the.plugin.KamonStarterPlugin
```

The other part of the issue is that the actor system is created when the ReactiveCouchbasePlugin is instanciated (https://github.com/ReactiveCouchbase/ReactiveCouchbase-play/blob/master/plugin/src/main/scala/org/reactivecouchbase/play/plugins/CouchbasePlugin.scala#L14).

So we need to change when the actor system is created and move it in the `onStart` function. Maybe you can submit a PR.

**Comment by chenbekor** [comment]
I would happily do that. can you please help me by explaining how to connect my existing app with a source code base of plugin? currently I'm linked to the remote repository binaries and if I'm making changes I want to test with my play app. waiting for you help.

**Comment by chenbekor** [comment]
fixed in 322891f1c5b8b69b0e6cb28092b66d9d2b04d4ca
